Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,653 @@
+				*/
+
+				void convertIfcPlane(const EXPRESSReference<typename IfcEntityTypesT::IfcPlane>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {
+
+					// Get basis surface.
+					EXPRESSReference<typename IfcEntityTypesT::IfcPlane>& basisSurface = surface->BasisSurface;
+
+					if (basisSurface) {
+
+						// Get basis surface position.
+						EXPRESSReference<typename IfcEntityTypesT::IfcAxis2Placement3D>& basisSurfacePlacement = basisSurface->Position;
+
+						if (basisSurfacePlacement) {
+							surfaceMatrix = pos * placementConverter->convertIfcAxis2Placement3D(basisSurfacePlacement);
 						}
 
-						throw oip::UnhandledException(surface);
 					}
+
+					// 1-----0	create big rectangular plane
+					// |	 |	^ y
+					// |	 |	|
+					// 2-----3	---> x
+
+					double planeSpan = HALF_SPACE_BOX_SIZE;
+					polylineData->beginPolyline();
+					polylineData->addVertex(surfaceMatrix * carve::geom::VECTOR(planeSpan, planeSpan, 0.0));
+					polylineData->addVertex(surfaceMatrix * carve::geom::VECTOR(-planeSpan, planeSpan, 0.0));
+					polylineData->addVertex(surfaceMatrix * carve::geom::VECTOR(-planeSpan, -planeSpan, 0.0));
+					polylineData->addVertex(surfaceMatrix * carve::geom::VECTOR(planeSpan, -planeSpan, 0.0));
+
+					polylineData->addPolylineIndex(0);
+					polylineData->addPolylineIndex(1);
+					polylineData->addPolylineIndex(2);
+					polylineData->addPolylineIndex(3);
+
 					throw oip::UnhandledException(surface);
+
 				}
 
-				// TODO: What is happening here?
-				void convertIfcFaceList(const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcFace>>& faces,
+				/*! \brief  Converts \c IfcSphericalPlane to ...
+				\param	surface	\c IfcSphericalPlane entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcSphericalPlane is a subtype of \c IfcElementarySurface.
+				*/
+
+				void convertIfcSphericalPlane(const EXPRESSReference<typename IfcEntityTypesT::IfcSphericalPlane>& surface,
 					const carve::math::Matrix& pos,
-					std::shared_ptr<ItemData> item_data
-				) const throw(...)
-				{
-					// carve polygon of the converted face list
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {
+
+					/* TO DO: Finish implementation of convertIfcSphericalPlane.
+
+					// Get radius.
+					double sphericalRadius = surface->Radius;
+
+					*/
+
+					throw oip::UnhandledException(surface);
+				}
+
+				/*! \brief  Converts \c IfcToroidalSurface to ...
+				\param	surface	\c IfcToroidalSurface entity to be interpreted.
+				\param	pos
+				\param	polylineData
+				\note	The \c IfcToroidalSurface is a subtype of \c IfcElementarySurface.
+				*/
+
+				void convertIfcToroidalSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcToroidalSurface>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...) {

```

Comment:
Same, maybe even both parameters.

In case the polyline data which is passed is not empty, rather return a new array and append that to the existing one.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/1ee08611e218ac8b03e6fabd0e214cad06a096b1. On request by @pjanck only polylineData as return parameter.